### DISCUSSION TITLE
noob to julia just starting with Stipple which init???

### DISCUSSION THREAD

**Discussion by ghost** [GenieFramework/Stipple.jl#125] [Q&A] [answered]
Hi there
decided to use the GermanCredits.jl to see what Stipple could do, included it into the repl and hit the
```
julia> include("GermanCredits.jl")
ERROR: LoadError: This method has been removed -- please use `init(Dashboard1; kwargs...)` instead
```
on the line
```
gc_model = setmodel(data, Dashboard1()) |> init
```

so being a noob I looked around and found

```
gc_model = Stipple.init( Dashboard1)
```
 It "fixes" the error BUT I want to make sure that this is the correct approach to Instantiating the Reactive model.

REMEMBER I am a Julia noob first so be gentle :-) 
theakson

**Comment by AbhimanyuAryan** [comment, marked as answer]
@sio2bagger yes that's the old Stipple syntax. To run the demo you can 

$ julia --project
julia> # ]
pkg> instantiate
julia> include("GermanCredits.jl")


```
(GermanCredits) pkg> instantiate
Precompiling project...
  3 dependencies successfully precompiled in 16 seconds (74 already precompiled)

julia> include("GermanCredits.jl")
┌ Warning: This method has been deprecated and will be removed soon. Please use `init(m::Type{M}, kwargs...)` instead.
└ @ Stipple ~/.julia/packages/Stipple/xELN6/src/Stipple.jl:792
┌ Info:
└ Web Server starting at http://127.0.0.1:8000
Genie.AppServer.ServersCollection(Task (runnable) @0x000000011d2e4230, nothing)

julia> [ Info: GET / 200
[ Info: GET /stipple.jl/master/assets/css/stipplecore.css 200
[ Info: GET /stippleui.jl/master/assets/css/quasar.min.css 200
[ Info: GET /genie.jl/master/assets/js/channels.js 200
[ Info: GET /stipple.jl/master/assets/js/underscore-min.js 200
[ Info: GET /stipple.jl/master/assets/js/vue.js 200
[ Info: GET /stipple.jl/master/assets/js/stipplecore.js 200
[ Info: GET /stipple.jl/master/assets/js/vue_filters.js 200
[ Info: GET /stipple.jl/master/assets/js/watchers.js 200
[ Info: GET /stipple.jl/master/assets/js/keepalive.js 200
[ Info: GET /stippleui.jl/master/assets/js/quasar.umd.min.js 200
[ Info: GET /stippleplotly.jl/master/assets/js/plotly2.min.js 200
[ Info: GET /stippleplotly.jl/master/assets/js/resizesensor.min.js 200
[ Info: GET /stippleplotly.jl/master/assets/js/lodash.min.js 200
[ Info: GET /stippleplotly.jl/master/assets/js/vueresize.min.js 200
[ Info: GET /stippleplotly.jl/master/assets/js/vueplotly.min.js 200
[ Info: GET /stipple.jl/master/assets/js/dashboard1.js 200
```

init(model) is used with newer versions of Stipple :)

**Reply by ghost** [reply]
thanks for taking the time to reply BUT not too sure what you mean.  I got it working with 
```
gc_model = Stipple.init( Dashboard1)
```
and you said that init(model) is not used any more.  

So what is the correct method please and where is it documented?  
Thanks so much for the EXCELLENT instructions as to how to run the latest code BUT being a NOOB to julia I don't know how to look at the code been run. Is it now on my machine or did "instantiate" point my environment at a new location.
thanks again for taking the time

**Reply by AbhimanyuAryan** [reply]
My bad. It's very late here. Sorry edited

**Reply by AbhimanyuAryan** [reply]
> thanks for taking the time to reply BUT not too sure what you mean.  I got it working with 
> ```
> gc_model = Stipple.init( Dashboard1)
> ```
> and you said that init(model) is not used any more.  
> 
> So what is the correct method please and where is it documented?  
> Thanks so much for the EXCELLENT instructions as to how to run the latest code BUT being a NOOB to julia I don't know how to look at the code been run. Is it now on my machine or did "instantiate" point my environment at a new location.
> thanks again for taking the time
> 

`instantiate` just installs the package versions given in Project.toml for your current environment i.e. GermanCredits

So its isolated from global package version of `julia` and reflect for current project(GermanCredits)

**Reply by ghost** [reply]
Hi there
thanks again BUT ( sorry about this) I still don't know what the correct call is to get the model loaded. So right now I am running my fix 

```
gc_model = Stipple.init( Dashboard1)
```
and it works. I thought you said it wasn't the current approach and I don't know what the new approach is. Please don't rush as it's late where you are.   I'll hunt around for the documentation.   
thanks again for taking the time.

**Reply by ghost** [reply]
I hunted around and found this documentation  https://docs.juliahub.com/Stipple/9RuwB/0.7.1/  which I am assuming is kept up to date.  In it they have
```
Base.@kwdef mutable struct Name <: ReactiveModel
  name::R{String} = "World!"
end

model = Stipple.init(Name())
```
do you agree?   

thanks again for taking the time to help a poor noob.
theakson

**Reply by AbhimanyuAryan** [reply]
No. You can also look at: [GenieFramework/StippleDemos](https://github.com/GenieFramework/StippleDemos) basic examples

**Reply by AbhimanyuAryan** [reply]
`Base.@kwdef`  is replaced with `@reactive`

**Reply by AbhimanyuAryan** [reply]
`|>` is used for method chaining so `init(Model)` and `Model |> init` are essentially the same thing

```
julia> sum([1:5;])
15

julia> [1:5;] |> sum
15
```
```
❯ julia +lts
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.6.6 (2022-03-28)
 _/ |\__'_|_|_|\__'_|  |  Official https://julialang.org/ release
|__/                   |

julia> using Stipple
[ Info: Precompiling Stipple [4acbeb90-81a0-11ea-1966-bdaff8155998]

julia> @reactive! mutable struct ChipModel <: ReactiveModel
       end
:ChipModel

julia> model = ChipModel |> init
ChipModel("JAGYYTCXWKSPFPVTSZPBRBVWOUTPSGEH", WARNING: both HttpCommon and Genie export "Headers"; uses of it in module Html must be qualified
WARNING: both Html and Base export "div"; uses of it in module Stipple must be qualified
WARNING: both Html and Base export "mark"; uses of it in module Stipple must be qualified
WARNING: both Html and Base export "summary"; uses of it in module Stipple must be qualified
WARNING: both Html and Base export "time"; uses of it in module Stipple must be qualified
WARNING: both Html and Base export "view"; uses of it in module Stipple must be qualified
Reactive{Bool}(Observable{Bool} with 1 listeners. Value:
false, 1, false, false), Reactive{Bool}(Observable{Bool} with 1 listeners. Value:
false, 1, false, false))
```

**Reply by ghost** [reply]
WOW what a GREAT set of answers. VERY much appreciated and I thank you for taking the time.  It's people like you who make a HUGE difference to people like me ( noobs). Thanks again.
theakson
